File: job_recruitment_app/lib/services/auth_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:shared_preferences/shared_preferences.dart';
import 'package:job_recruitment_app/constants/api_constants.dart';
import 'package:job_recruitment_app/models/user.dart';

class AuthService {
  static Future<User?> register({
    required String email,
    required String password,
    required String fullName,
    required String userType,
    String? phone,
  }) async {
    try {
      final response = await http.post(
        Uri.parse('${ApiConstants.baseUrl}${ApiConstants.register}'),
        headers: {'Content-Type': 'application/json'},
        body: json.encode({
          'email': email,
          'password': password,
          'fullName': fullName,
          'userType': userType,
          'phone': phone,
        }),
      );

      if (response.statusCode == 200) {
        final data = json.decode(response.body);
        final user = User.fromJson(data);
        
        // Save user data
        final prefs = await SharedPreferences.getInstance();
        await prefs.setString('token', user.token);
        await prefs.setString('user', json.encode(user.toJson()));
        
        return user;
      }
      return null;
    } catch (e) {
      print('Register error: $e');
      return null;
    }
  }

  static Future<User?> login({
    required String email,
    required String password,
  }) async {
    try {
      final response = await http.post(
        Uri.parse('${ApiConstants.baseUrl}${ApiConstants.login}'),
        headers: {'Content-Type': 'application/json'},
        body: json.encode({
          'email': email,
          'password': password,
        }),
      );

      if (response.statusCode == 200) {
        final data = json.decode(response.body);
        final user = User.fromJson(data);
        
        // Save user data
        final prefs = await SharedPreferences.getInstance();
        await prefs.setString('token', user.token);
        await prefs.setString('user', json.encode(user.toJson()));
        
        return user;
      }
      return null;
    } catch (e) {
      print('Login error: $e');
      return null;
    }
  }

  static Future<void> logout() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.clear();
  }

  static Future<User?> getCurrentUser() async {
    final prefs = await SharedPreferences.getInstance();
    final userString = prefs.getString('user');
    
    if (userString != null) {
      final data = json.decode(userString);
      return User.fromJson(data);
    }
    return null;
  }
}

File: job-recruitment-frontend/src/services/authService.ts
import api from './api';
import { setToken, setUser, clearAuth, getToken, getUser } from '../utils/helpers';
import type { AuthResponse, LoginData, RegisterData, User } from '../types/user';


class AuthService {
  async register(data: RegisterData): Promise<User> {
    try {
      const response = await api.post<AuthResponse>('/auth/register', data);
      setToken(response.token);
      const user: User = {
        id: response.id,
        email: response.email,
        fullName: response.fullName,
        userType: response.userType as 'Recruiter' | 'Seeker',
        token: response.token,
      };
      setUser(user);
      return user;
    } catch (error) {
      clearAuth();
      throw error;
    }
  }

  async login(data: LoginData): Promise<User> {
    try {
      const response = await api.post<AuthResponse>('/auth/login', data);
      setToken(response.token);
      const user: User = {
        id: response.id,
        email: response.email,
        fullName: response.fullName,
        userType: response.userType as 'Recruiter' | 'Seeker',
        token: response.token,
      };
      setUser(user);
      return user;
    } catch (error) {
      clearAuth();
      throw error;
    }
  }

  logout(): void {
    clearAuth();
  }

  getCurrentUser(): User | null {
    return getUser();
  }

  isAuthenticated(): boolean {
    return !!getToken();
  }

  isRecruiter(): boolean {
    const user = this.getCurrentUser();
    return user?.userType === 'Recruiter';
  }

  isSeeker(): boolean {
    const user = this.getCurrentUser();
    return user?.userType === 'Seeker';
  }
}

export default new AuthService();

